### ISSUE TITLE
Gitlab-Pipeline Crash after new release 2022.11.4

### ISSUE THREAD

**Issue by petrroda** [pypa/pipenv#5459]
Our build crashes with the new release of pipenv:

Collecting pipenv
  Downloading pipenv-2022.11.4-py2.py3-none-any.whl (2.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.9/2.9 MB [42](https://gitlab.newworkjob.de/gitlab-instance-31e781f1/frontend/-/jobs/1632#L42).9 MB/s eta 0:00:00
Collecting virtualenv-clone>=0.2.5
  Downloading virtualenv_clone-0.5.7-py3-none-any.whl (6.6 kB)
Collecting certifi
  Downloading certifi-2022.9.24-py3-none-any.whl (161 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 161.1/161.1 KB 32.1 MB/s eta 0:00:00
Requirement already satisfied: setuptools>=36.2.1 in /usr/local/lib/python3.8/site-packages (from pipenv) (57.5.0)
Collecting virtualenv
  Downloading virtualenv-20.16.6-py3-none-any.whl (8.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8.8/8.8 MB 65.5 MB/s eta 0:00:00
Collecting platformdirs<3,>=2.4
  Downloading platformdirs-2.5.2-py3-none-any.whl (14 kB)
Collecting filelock<4,>=3.4.1
  Downloading filelock-3.8.0-py3-none-any.whl (10 kB)
Collecting distlib<1,>=0.3.6
  Downloading distlib-0.3.6-py2.py3-none-any.whl ([46](https://gitlab.newworkjob.de/gitlab-instance-31e781f1/frontend/-/jobs/1632#L46)8 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.5/468.5 KB 75.2 MB/s eta 0:00:00
Installing collected packages: distlib, virtualenv-clone, platformdirs, filelock, certifi, virtualenv, pipenv
Successfully installed certifi-2022.9.24 distlib-0.3.6 filelock-3.8.0 pipenv-2022.11.4 platformdirs-2.5.2 virtualenv-20.16.6 virtualenv-clone-0.5.7
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
WARNING: You are using pip version 22.0.4; however, version 22.3 is available.
You should consider upgrading via the '/usr/local/bin/python -m pip install --upgrade pip' command.
INFO[0012] Taking snapshot of full filesystem...        
INFO[0014] ENV PROJECT_DIR /app                         
INFO[0014] COPY ./src /${PROJECT_DIR}                   
INFO[0014] Taking snapshot of files...                  
INFO[0014] COPY ./Pipfile /${PROJECT_DIR}               
INFO[0014] Taking snapshot of files...                  
INFO[0014] RUN rm -rf ${PROJECT_DIR}/.env               
INFO[0014] Cmd: /bin/sh                                 
INFO[0014] Args: [-c rm -rf ${PROJECT_DIR}/.env]        
INFO[0014] Running: [/bin/sh -c rm -rf ${PROJECT_DIR}/.env] 
INFO[0014] Taking snapshot of full filesystem...        
INFO[0015] No files were changed, appending empty layer to config. No layer added to image. 
INFO[0015] WORKDIR ${PROJECT_DIR}                       
INFO[0015] Cmd: workdir                                 
INFO[0015] Changed working directory to /app            
INFO[0015] No files changed in this command, skipping snapshotting. 
INFO[0015] RUN pipenv lock                              
INFO[0015] Cmd: /bin/sh                                 
INFO[0015] Args: [-c pipenv lock]                       
INFO[0015] Running: [/bin/sh -c pipenv lock]            
Creating a virtualenv for this project...
Pipfile: /app/Pipfile
Using /usr/local/bin/python3.8 (3.8.15) to create virtualenv...
created virtual environment CPython3.8.15.final.0-64 in 811ms
  creator Venv(dest=/root/.local/share/virtualenvs/app-4PlAip0Q, clear=False, no_vcs_ignore=False, global=False, describe=CPython3Posix)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/root/.local/share/virtualenv)
    added seed packages: pip==22.3, setuptools==65.5.0, wheel==0.37.1
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/pipenv/utils/spinner.py", line 18, in create_spinner
    yield sp
  File "/usr/local/lib/python3.8/site-packages/pipenv/core.py", line 1019, in do_create_virtualenv
    sp.green.ok(
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/vistir/spin.py", line 264, in ok
    self._freeze(_text, err=err)
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/vistir/spin.py", line 363, in _freeze
    last_frame = self._compose_out(text, mode="last")
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/vistir/spin.py", line 386, in _compose_out
    if self._color_func is not None:
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/vistir/spin.py", line 115, in __getattr__
    retval = super(DummySpinner, self).__getattribute__(k)
AttributeError: 'VistirSpinner' object has no attribute '_color_func'
Traceback (most recent call last):
  File "/usr/local/bin/pipenv", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/pipenv/cli/options.py", line 57, in main
    return super().main(*args, **kwargs, windows_expand_args=False)
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/click/core.py", line 10[53](https://gitlab.newworkjob.de/gitlab-instance-31e781f1/frontend/-/jobs/1632#L53), in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/click/core.py", line 7[54](https://gitlab.newworkjob.de/gitlab-instance-31e781f1/frontend/-/jobs/1632#L54), in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/pipenv/cli/command.py", line 328, in lock
    ensure_project(
  File "/usr/local/lib/python3.8/site-packages/pipenv/core.py", line 530, in ensure_project
    ensure_virtualenv(
  File "/usr/local/lib/python3.8/site-packages/pipenv/core.py", line 4[63](https://gitlab.newworkjob.de/gitlab-instance-31e781f1/frontend/-/jobs/1632#L63), in ensure_virtualenv
    do_create_virtualenv(
  File "/usr/local/lib/python3.8/site-packages/pipenv/core.py", line 1019, in do_create_virtualenv
    sp.green.ok(
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/vistir/spin.py", line 2[64](https://gitlab.newworkjob.de/gitlab-instance-31e781f1/frontend/-/jobs/1632#L64), in ok
    self._freeze(_text, err=err)
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/vistir/spin.py", line 363, in _freeze
    last_frame = self._compose_out(text, mode="last")
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/vistir/spin.py", line 3[86](https://gitlab.newworkjob.de/gitlab-instance-31e781f1/frontend/-/jobs/1632#L86), in _compose_out
    if self._color_func is not None:
  File "/usr/local/lib/python3.8/site-packages/pipenv/vendor/vistir/spin.py", line 115, in __getattr__
    retval = super(DummySpinner, self).__getattribute__(k)
AttributeError: 'VistirSpinner' object has no attribute '_color_func'
error building image: error building stage: failed to execute command: waiting for process to exit: exit status 1

**Comment by alancummings** [comment]
I'm getting the same error.
Environment is within a docker build (building fails)
relevant Dockerfile lines:
```
FROM python:3.8

RUN cd <directory> \
  && pip install pip --upgrade \
  && pip install pipenv --upgrade \
  && pipenv install
```

**Comment by matteius** [comment]
Can someone else with the issue help to verify the fix of the PR I opened #5460 
I am pretty confident that is is, but a second 👍 would be good -- and I can cut a release after that.
